austation/code/modules/coilgun/charger.dm

```diff
@@ -0,0 +1,152 @@
+	var/cps = 0 // current projectile speed, stored in a var for examining the charger
+	var/list/members = list()
+	var/parent = null // used for linking coilgun chargers, what charger is parent?
+	var/is_child = FALSE // is this linked to a parent?
+	var/laststep // used in charger chain building, stops infinite loops. did we just run build proc on this pipe?
+
+// needed for charger linking
+/obj/structure/disposalpipe/coilgun/charger/New()
+	parent = src
+	members += src
+
+// because I don't want to make a GUI
+/obj/structure/disposalpipe/coilgun/charger/attack_hand(mob/user)
+	. = ..()
+	if(.)
+		return
+	if(enabled)
+		if(check_connection(parent))
+			if(target_power_usage >= 100)
+				enabled = FALSE
+				target_power_usage = 0
+				to_chat(user, "<span class='notice'>You turn off \the [src].</span>")
+			else
+				target_power_usage += 20
+				to_chat(user, "<span class='notice'>You increase \the [src]'s power throttle to [target_power_usage]%</span>")
+		else
+			to_chat(user, "<span class='warning'>\The [src] has no power!</span>")
+	else
+		enabled = TRUE
+		to_chat(user, "<span class='notice'>You turn on \the [src].</span>")
+
+	if(members.len <= 1 && parent == src) // if it's not a child or parent of another object, try to sync with nearvy chargers
+		build_charger(parent)
+	update_chargers(parent) //sync the connected charger's settings
+
+/// updates the chargers connected to the parent
+/obj/structure/disposalpipe/coilgun/charger/proc/update_chargers(obj/structure/disposalpipe/coilgun/charger/P)
+	for(var/obj/structure/disposalpipe/coilgun/charger/C in P.members)
+		C.target_power_usage = target_power_usage
+		C.enabled = enabled
+		C.attached = attached // only one cable needs to be attached to a charger
+
+/// Finds all chargers connected to the caller (parent) and makes them members
+/obj/structure/disposalpipe/coilgun/charger/proc/build_charger(obj/structure/disposalpipe/coilgun/charger/P)
+	var/obj/structure/disposalpipe/coilgun/charger/C
+	for(var/turf/T in range(1, src)) // for every tile next to the charger
+		C = locate() in T
+		if(C && C.dpdir == P.dpdir && (!C.parent || C.parent == C) && C.laststep != laststep)
+			if(laststep == C)
+				continue
+			laststep = src
+			if(!(C in P.members))
+				P.members += C
+				C.parent = P
+			C.visible_message("<span class='warning'>Debug: building charger...</span>")
+			C.build_charger(P)
+			C.visible_message("<span class='warning'>Debug: synced with parent!</span>")
+
+/obj/structure/disposalpipe/coilgun/charger/proc/check_connection(obj/structure/disposalpipe/coilgun/charger/P)
+	for(var/obj/structure/disposalpipe/coilgun/charger/C in P.members)
+		var/turf/T = loc
+		if(isturf(T) && !T.intact)
+			attached = locate() in T
+			if(attached)
+				return TRUE
+
+	return FALSE
+
+/obj/structure/disposalpipe/coilgun/charger/proc/power_process()
+	if(current_power_use && target_power_usage)
+		var/datum/powernet/PN = attached.powernet
+		if(attached && PN)
```

Comment:
I can spot a potential null runtime on line 91. You need to make sure attached actually exists *before* trying to retrieve a powernet from it.